internal/cli/usercmd/push.go

```diff
@@ -268,24 +268,38 @@ func (c *EpinioClient) AppPush(ctx context.Context, manifest models.ApplicationM
 func (c *EpinioClient) uploadSources(log logr.Logger, appRef models.AppRef, source string) (string, error) {
 	c.ui.Normal().Msg("Collecting the application sources ...")
 
-	tmpDir, tarball, err := helpers.Tar(source)
-	defer func() {
-		if tmpDir != "" {
-			_ = os.RemoveAll(tmpDir)
-		}
-	}()
+	fileInfo, err := os.Stat(source)
 	if err != nil {
 		return "", err
 	}
 
+	var archive string
+
+	if fileInfo.IsDir() {
+		// package directory as archive/tarball
+		tmpDir, tarball, err := helpers.Tar(source)
+		defer func() {
+			if tmpDir != "" {
+				_ = os.RemoveAll(tmpDir)
+			}
+		}()
+		if err != nil {
+			return "", err
+		}
+		archive = tarball
+	} else {
+		// consider file as the archive
+		archive = source
+	}
```

Comment:
If in the `else` we have only this we can remove it completely initializing the `archive` as `source` at the beginning, and then if `fileInfo.IsDir` we can overwrite it:

```go
archive := source

if fileInfo.IsDir() {
	// blabla
	archive = tarball
}

```

Reply:
Done. Commit 450458216cfa6b5b756f082194f3a26a246a33c9